aiohttp_jinja2/__init__.py

```diff
@@ -86,7 +86,12 @@ def wrapped(*args):
             if isinstance(context, web.StreamResponse):
                 return context
 
-            request = args[-1]
+            # Supports class based views see web.View
+            if isinstance(args[0], web.View):
```

Comment:
Good call, I should have looked more at the source - I was a little too rushed.  Thanks!